src/parser.yy

```diff
@@ -219,8 +219,8 @@ stmts : semicolon_ended_stmt stmts { $$ = $2; $2->insert($2->begin(), $1); }
       |                            { $$ = new ast::StatementList; }
       ;
 
-block_stmt : if_stmt                  { $$ = $1; }
-           | UNROLL "(" INT ")" block { $$ = new ast::Unroll($3, $5); }
+block_stmt : if_stmt                           { $$ = $1; }
+           | UNROLL "(" int_or_param ")" block { $$ = new ast::Unroll($3, $5, @1 + @4); }
```

Comment:
```suggestion
           | UNROLL "(" int ")" block { $$ = new ast::Unroll($3, $5, @1 + @4); }
            | UNROLL "(" param ")" block { $$ = new ast::Unroll($3, $5, @1 + @4); }
```

Currently we have the `INT` (lexer) token and the `int` symbol in the parser which handles negative ints too. This would be a `positive_int_or_param`, which I think would add more confusion than it solves. Additionally this moves the detection of negative numbers to the semantic analyser, which can produce nicer errors:

```
stdin:1:17-28: ERROR: unroll minimum value is 1
i:s:1 { $a= 0 ; unroll(-10) { printf("%d\n", $a++); } exit() }
                ~~~~~~~~~~~
```

Compared to

```
stdin:1:17-25: ERROR: syntax error, unexpected -, expecting $# or positional parameter or integer
i:s:1 { $a= 0 ; unroll(-10) { printf("%d\n", $a++); } exit() }
                ~~~~~~~~
```